sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/options/ShareDirectoryListFilesAndDirectoriesOptions.java

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file.share.options;
+
+import com.azure.storage.file.share.models.ListFilesIncludeType;
+
+import java.util.EnumSet;
+
+/**
+ * Extended options for a directory listing operation.
+ */
+public class ShareDirectoryListFilesAndDirectoriesOptions {
+
+    private String prefix;
+    private Integer maxResultsPerPage;
+    private EnumSet<ListFilesIncludeType> shareFileTraits;
```

Comment:
I wonder if we should follow the pattern of listBlobs instead here? See BlobContainerListDetails